StyleCop.Analyzers/StyleCop.Analyzers.Test.CSharp8/Lightup/SwitchExpressionArmSyntaxWrapperTests.cs

```diff
@@ -0,0 +1,118 @@
+
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.WithExpression(SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression)));
+            Assert.Throws<NullReferenceException>(() => switchExpressionSyntax.WithEqualsGreaterThanToken(SyntaxFactory.Token(SyntaxKind.EqualsGreaterThanToken)));
+        }
+
+        [Fact]
+        public void TestProperties()
+        {
+            var syntaxNode = this.CreateSwitchExpressionArm();
+
+            Assert.True(syntaxNode.IsKind(SyntaxKind.SwitchExpressionArm));
+            Assert.True(syntaxNode.IsKind(SyntaxKindEx.SwitchExpressionArm));
+
+            var wrapper = (SwitchExpressionArmSyntaxWrapper)syntaxNode;
+            Assert.Same(syntaxNode, wrapper.SyntaxNode);
+            Assert.Same(syntaxNode.Pattern, wrapper.Pattern.SyntaxNode);
+            Assert.Same(syntaxNode.WhenClause, wrapper.WhenClause.SyntaxNode);
+            Assert.Same(syntaxNode.Expression, wrapper.Expression);
+            Assert.Equal(syntaxNode.EqualsGreaterThanToken, wrapper.EqualsGreaterThanToken);
+
+            // Pattern
+            var newPattern = SyntaxFactory.ConstantPattern(SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression));
+            var wrapperWithModifiedPattern = wrapper.WithPattern((PatternSyntaxWrapper)newPattern);
+            Assert.NotNull(wrapperWithModifiedPattern.SyntaxNode);
+            Assert.NotSame(syntaxNode.Pattern, wrapperWithModifiedPattern.Pattern.SyntaxNode);
+            Assert.Equal(SyntaxKind.ConstantPattern, wrapperWithModifiedPattern.Pattern.SyntaxNode.Kind());
+
+            // When Clause
+            var newWhenClause = SyntaxFactory.WhenClause(SyntaxFactory.LiteralExpression(SyntaxKind.TrueLiteralExpression));
+            var wrapperWithModifiedWhenClause = wrapper.WithWhenClause((WhenClauseSyntaxWrapper)newWhenClause);
+            Assert.NotNull(wrapperWithModifiedWhenClause.SyntaxNode);
+            Assert.NotSame(syntaxNode.WhenClause, wrapperWithModifiedWhenClause.WhenClause.SyntaxNode);
+            Assert.Equal(SyntaxKind.WhenClause, wrapperWithModifiedWhenClause.WhenClause.SyntaxNode.Kind());
+            Assert.Equal(SyntaxKind.TrueLiteralExpression, wrapperWithModifiedWhenClause.WhenClause.Condition.Kind());
+
+            // Expression
+            var newExpression = SyntaxFactory.LiteralExpression(SyntaxKind.NullLiteralExpression);
+            var wrapperWithModifiedExpression = wrapper.WithExpression(newExpression);
+            Assert.NotNull(wrapperWithModifiedExpression.SyntaxNode);
+            Assert.NotSame(syntaxNode.Expression, wrapperWithModifiedExpression.Expression);
+            Assert.Equal(SyntaxKind.NullLiteralExpression, wrapperWithModifiedExpression.Expression.Kind());
+
+            // EqualsGreaterThanToken
+            Assert.Throws<ArgumentException>(() => wrapper.WithEqualsGreaterThanToken(SyntaxFactory.Token(SyntaxKind.EqualsEqualsToken)));
```

Comment:
❔ Is it possible to test this with a passing case?

Reply:
Since this will throw if the token is `!= SyntaxKind.EqualsGreaterThanToken` the only solution I see is to pass this as a parameter. It's not a perfect test since it has already the same value but at least we can see that it doesn't throw.

Reply:
You can create a token with this kind, but set the leading or trailing trivia to something new and test that. For example, `/*1*/=>` would be semantically the same as `=>` but is still different enough to observe in a test. 👍